lib/Doctrine/ORM/QueryBuilder.php

```diff
@@ -1076,7 +1076,9 @@ public function set($key, $value)
      *         ->where($or);
      * </code>
      *
-     * @param mixed $predicates The restriction predicates.
+     * @param                 string|object|array $predicates The restriction predicates.
+     * @psalm-param   literal-string|object|array $predicates
```

Comment:
Doesn't Psalm also read the phpstan annotations? You can probably omit the psalm one as it might be redundant.

Reply:
Yes, you're right, [psalm does read phpstan annotations](https://github.com/vimeo/psalm/blob/af8033adf037017c7088da773ec6b8bb7711d741/src/Psalm/Internal/Scanner/DocblockParser.php#L231) (I had no idea, thanks). While it is technically redundant at the moment, maybe it makes it clear that it's supported by both psalm and phpstan; it's also possible psalm could stop doing this (although, if that does happen, we can probably move to using `@psalm literal-string` for everyone, [when phan supports it](https://github.com/phan/phan/issues/4618)?). For reference, Mark Scherer preferred to list both for Propel2.

What would you prefer?

Reply:
For maintenance it's easier to have only one of them, which is the phpstan one.

Reply:
That's fair, @psalm-param has been removed, and a quick test confirms it still works.